packages/gatsby-remark-prismjs/src/add-line-numbers.js

```diff
@@ -8,7 +10,11 @@ module.exports = (code = []) => {
     return spans
   }
 
-  const numberOfLines = code.length === 0 ? 0 : code.split(`\n`).length
+  const numberOfLines =
+    code.length === 0
+      ? 0
+      : code.split(`\n`).length +
```

Comment:
now that we operate on html - does splitting on `\n` is actually what we want? Don't remember right now if we use `white-space: pre` on code blocks to have line breaks in rendered content or we use blocks (i.e. `<div>`) or explicit line breaks (`<br/>`) for it

Reply:
I just tested this locally and it seems [`highlightCode.js`](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-remark-prismjs/src/highlight-code.js) (near the bottom) uses `\n` to line break everything including the line breaks specified within the input content.

This was what I was initially working off when making the changes that piped the output to this new line calculation. 

Is this what you meant or is there another scenario where this would not be the case?

Input:
```js
function highlightTest() {


  // highlight-start
  return "this is a highlight test"
  // highlight-end
}
```
Code splits (before output of `highlightCode.js`):
```
[
  {
    code: '<span class="token keyword">function</span> <span class="token function">highlightTest</span><span class="token punctuation">(</span><span class="token punctuation">)</span> <span class="token punctuation">{</span>',
    highlight: false,
    hide: false,
    flagSources: []
  },
  { code: '', highlight: false, hide: false, flagSources: [] },
  { code: '', highlight: false, hide: false, flagSources: [] },
  {
    code: '<span class="gatsby-highlight-code-line">  <span class="token keyword">return</span> <span class="token string">"this is a highlight test"</span></span>',
    highlight: true,
    hide: false,
    flagSources: [ [Object] ]
  },
  {
    code: '<span class="token punctuation">}</span>',
    highlight: false,
    hide: false,
    flagSources: []
  }
]
```
Output of `highlightCode.js`
```
<span class="token keyword">function</span> <span class="token function">highlightTest</span><span class="token punctuation">(</span><span class="token punctuation">)</span> <span class="token punctuation">{</span>


<span class="gatsby-highlight-code-line">  <span class="token keyword">return</span> <span class="token string">"this is a highlight test"</span></span><span class="token punctuation">}</span>
```